### PR TITLE
update event skeleton to new input event file structure

### PR DESCRIPTION
**Pull request by cg-laser**
the event generator skeleton was not updated yet to the new hdf5 file structure. This is fixed with the PR.

### CODE REVIEW

**Comment by cg-laser** [comment]
I also added that the weights can be specified already in the evt gen

File: NuRadioMC/simulation/simulation.py [hunk 1]
```diff
@@ -432,7 +432,10 @@ def run(self):
             t1 = time.time()
             iE_mother = event_indices[0]
             x_int_mother = np.array([self._fin['xx'][iE_mother], self._fin['yy'][iE_mother], self._fin['zz'][iE_mother]])
-            self._mout['weights'][event_indices] = get_weight(self._fin['zeniths'][iE_mother],
+            if("weights" in self._fin):
+                self._mout['weights'] = self._fin["weights"]
+            else:
```
Comment (shallmann) [inline NuRadioMC/simulation/simulation.py:437]: In the past I have looked at the attrs of a hdf5 file to see which weights are in the file. Is it still guaranteed that the  information written in `self._cfg['weights']['weight_mode']` and `self._cfg['weights']['cross_section_type']` will be correct?

Would be annoying if one needs to reverse engineer and verify what has really been run by re-calculating event weights by hand... ;)

File: NuRadioMC/EvtGen/generator_skeleton.py [hunk 2]
```diff
@@ -30,37 +31,72 @@ def generate_my_events(filename, n_events):
...
 
-    data_sets["interaction_type"] = np.ones(n_events, dtype='S2')
-    data_sets["inelasticity"] = np.ones(n_events) *  0.5
+    # optionally one can also directly set the event weight here (useful if particles other than neutrinos, or calibration
+    # setups are simulated
+    # data_sets["weights"] = np.ones(n_events)
 
+    # the interaction type. For neutrino interactions is can be either CC or NC. This parameter is not used but passed
+    # to the output file for information purposes.
+    data_sets["interaction_type"] = np.ones(n_events, dtype='S2')
```
Comment (shallmann) [inline NuRadioMC/EvtGen/generator_skeleton.py:96]: generally for the skeleton, np.ones/np.zeros are perfectly fine. For the 2 letter strings, maybe it is more intuitive to fill it with one of the actual values?

i.e. `np.ones(n_events, dtype='S2')` -> `np.full(n_events, 'CC', dtype='S2')`

**Review by shallmann — CHANGES_REQUESTED** [review]
hesitant with approval concerning my comment on the attributes, want to hear your opinion on this first.

File: NuRadioMC/EvtGen/generator_skeleton.py (hunk 2 above)
Comment (cg-laser) [reply NuRadioMC/EvtGen/generator_skeleton.py:96]: good point, will change that.

Comment (shallmann) [reply NuRadioMC/EvtGen/generator_skeleton.py]: @cg-laser another comment:

I guess fixing this resolves the future unimplemented warning when generating a terminal dump of the file:

/Users/shallmann/miniconda/lib/python3.8/site-packages/tables/attributeset.py:308: DataTypeWarning: Unsupported type for attribute 'deposited' in node '/'. Offending HDF5 class: 8
  value = self._g_getattr(self._v_node, name)
/ (RootGroup) ''
/Users/shallmann/miniconda/lib/python3.8/site-packages/tables/group.py:1201: UserWarning: problems loading leaf ``/interaction_type``::

  variable length strings are not supported yet

The leaf will become an ``UnImplemented`` node.
  warnings.warn(
/Users/shallmann/miniconda/lib/python3.8/site-packages/tables/group.py:1201: UserWarning: problems loading leaf ``/shower_type``::

  variable length strings are not supported yet

The leaf will become an ``UnImplemented`` node.

Comment (shallmann) [reply NuRadioMC/EvtGen/generator_skeleton.py]: but do we actually only ever store CC or NC there? If I remember correctly there was also a "tau_had", "tau_em" at some place stored? Or was that in another field?

Comment (cg-laser) [reply NuRadioMC/EvtGen/generator_skeleton.py]: at the moment we don't use it, but we might in the future. The thing is, this field is not used at all in the simulation but you still want to know it for post analyses. In the secondary shower generation, one could use this field to save what type of interaction happend (and I think this is already being done). What matters for the simulation is the "shower_type" which is so far only "em" or "had". But in principle there could be more channels, e.g. a tau decay is slightly differnt.

**Comment by cg-laser** [comment]
@christophwelling does this look good?

File: NuRadioMC/EvtGen/generator_skeleton.py [hunk 3]
```diff
@@ -30,37 +24,91 @@ def generate_my_events(filename, n_events):
...
+    # the direction of the shower
     data_sets["azimuths"] = np.ones(n_events)
     data_sets["zeniths"] = np.ones(n_events)
+    # the position of the shower
     data_sets["xx"] = np.ones(n_events)
     data_sets["yy"] = np.ones(n_events)
     data_sets["zz"] = np.ones(n_events)
-    data_sets["event_ids"] = np.arange(n_events)
+    # the shower energy
+    data_sets["shower_energies"] = np.ones(n_events) * 1 * units.EeV
```
Comment (christophwelling) [inline NuRadioMC/EvtGen/generator_skeleton.py:73]: Why are showers initialized with 1EeV?

**Review by christophwelling — DISMISSED** [review]
Looks alright, but I don't understand why some of the values are specified with specific values. I guess it's just as an example? Then a docstring explaining that would be good.

File: NuRadioMC/EvtGen/generator_skeleton.py (hunk 3 above)
Comment (shallmann) [reply NuRadioMC/EvtGen/generator_skeleton.py:73]: this is a valid point. We probably shouldn't suggest to set shower energies manually at all? Is it less error prone, to provide external functions for generating an energy spectrum from the attributes?... like a powerlaw between Emin, Emax (ok we don't store an index (yet)). (don't know if this unnecessarily overcomplicate things for us, the optical telescopes typically use powerlaws between emin/max in the generator)

File: NuRadioMC/EvtGen/generator_skeleton.py (hunk 2 above)
Comment (shallmann) [reply NuRadioMC/EvtGen/generator_skeleton.py]: ok, so at least for the shower type we should make sure the string length is long enough, like `dtype='S7'`?

File: NuRadioMC/EvtGen/generator_skeleton.py (hunk 3 above)
Comment (cg-laser) [reply NuRadioMC/EvtGen/generator_skeleton.py:73]: this skeleton is intended for non standard use cases. So manually setting shower energies is something that people need to know how to do

File: NuRadioMC/simulation/simulation.py (hunk 1 above)
Comment (shallmann) [reply NuRadioMC/simulation/simulation.py:443]: looks better to me

**Review by shallmann — APPROVED** [review]
The change to use non-standard weights looks good to me now. I have no further objections.

Acutally I have one small comment: Although it is meant as a custom dummy script and the user is responsible for filling the simulations with useful data, I wouldn't put neutrino energy AND shower energy to 1 EeV and set the Bjorken y to 0.5 which is obviously wrong!

**Comment by cg-laser** [comment]
we use this custom event generator for example to generate simulations for air-shower cores hitting the surface. There we need the option to directly set the shower energies. There are quite some non-neutrino use cases where inelasticity and total energy are not used.

**Review by shallmann — APPROVED** [review]
fine with me now

**Pull request merged**